### DISCUSSION TITLE
modifying html5 IP Geolocation provider

### DISCUSSION THREAD

**Discussion by tech-maven-geospatial** [TerriaJS/terriajs#4933] [Q&A] [answered]
We have developed a WPF - Windows Presentation Foundation Desktop Application for Terriajs
 with lots of Native C# .NET code 
running local webserver and Chromium for the JavaScript V8 Engine and WebGL Rendering
and local tile server/provider for offline mbtiles/gpkg and 3DTiles
I am having issues figuring out what file to modify for the Map Geolocation

We have an internal api 

just send GET request to
http://localhost:5050/GetMyLocation
returns JSON

{"latitude":41.076711652269488,"longitude":29.002793221387442,"isSuccess":true,"message":null}

That I want to geolocation button to work
What file do we need to modify to get this working?
is it https://github.com/TerriaJS/terriajs/blob/master/lib/ReactViews/Map/Navigation/MyLocation.jsx

**Comment by tephenavies** [comment, marked as answer]
Hi @tech-maven-geospatial. That should be the only file you need to modify. Replace `navigator.geolocation` API calls with your own GET requests (terriajs' `loadJson` file/function could be convenient).
